opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/websocket/TextMapExtractAdapter.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.websocket;
+
+import io.opentracing.Tracer;
+import io.opentracing.propagation.TextMap;
+
+import java.util.HashMap;
+import java.util.Iterator;
+
+import java.util.Map;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * A TextMap carrier for use with Tracer.extract() to extract tracing context from Spring messaging
+ * {@link MessageHeaders}.
+ *
+ * @see Tracer#extract(Format, Object)
```

Comment:
I have commented before: this is an invalid javadoc you have to import format. The same is in inject adapter.